src/components/structures/MessagePanel.js

```diff
@@ -292,16 +308,18 @@ module.exports = React.createClass({
 
         for (i = 0; i < this.props.events.length; i++) {
             let mxEv = this.props.events[i];
-            let wantTile = true;
             let eventId = mxEv.getId();
             let readMarkerInMels = false;
+            let last = (i == lastShownEventIndex);
 
-            if (!EventTile.haveTileForEvent(mxEv)) {
-                wantTile = false;
+            if (!this._shouldShowEvent(mxEv)) {
+                // Event is hidden but may be the read marker event
+                if (mxEv.getId() === this.props.readMarkerEventId) {
+                    ret.push(this._getReadMarkerTile(this.props.readMarkerVisible));
+                }
+                continue;
```

Comment:
oh. This is the bug, there just shouldn't be a `continue` here. And we should still be using `wantTile`.